Userland/Services/Taskbar/ClockWidget.cpp

```diff
@@ -226,4 +231,22 @@ void ClockWidget::jump_to_current_date()
     m_selected_calendar_button->set_text(m_calendar->formatted_date());
 }
 
+int ClockWidget::time_width()
+{
+    auto width_string = m_time_format_to_width_string.get(m_time_format);
+    return font().width(width_string.value());
+}
+
+void ClockWidget::set_time_format(String const& format)
+{
+    m_time_format = format;
+    set_fixed_size(time_width() + 20, 21);
+}
+
+String ClockWidget::time_format()
```

Comment:
The current names sounds like it obtains which time format to print in, when it's really performing the action of formatting the time. Maybe a name like this would be better? 

```suggestion
String ClockWidget::fetch_and_format_time()
```